### ISSUE TITLE
[Installation]: install fails on Ubuntu 24.04

### ISSUE THREAD

**Issue by calz1** [PygmalionAI/aphrodite-engine#514]
### Your current environment

```text
python env.py
--2024-06-16 20:27:30--  https://raw.githubusercontent.com/PygmalionAI/aphrodite-engine/main/env.py
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 2606:50c0:8001::154, 2606:50c0:8002::154, 2606:50c0:8003::154, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|2606:50c0:8001::154|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 20450 (20K) [text/plain]
Saving to: ‘env.py’

env.py                        100%[================================================>]  19.97K  --.-KB/s    in 0s

2024-06-16 20:27:31 (112 MB/s) - ‘env.py’ saved [20450/20450]

Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A
OS: Ubuntu 24.04 LTS (x86_64)
GCC version: (Ubuntu 13.2.0-23ubuntu4) 13.2.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.39
Python version: 3.12.3 (main, Apr 10 2024, 05:33:47) [GCC 13.2.0] (64-bit runtime)
Python platform: Linux-6.8.0-35-generic-x86_64-with-glibc2.39
Is CUDA available: N/A
CUDA runtime version: 12.0.140
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration:
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090
GPU 2: NVIDIA GeForce RTX 3090
GPU 3: NVIDIA GeForce RTX 3090

Nvidia driver version: 535.171.04
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A
CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        48 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               12
On-line CPU(s) list:                  0-11
Vendor ID:                            AuthenticAMD
Model name:                           AMD Ryzen 5 5500GT with Radeon Graphics
CPU family:                           25
Model:                                80
Thread(s) per core:                   2
Core(s) per socket:                   6
Socket(s):                            1
Stepping:                             0
CPU(s) scaling MHz:                   57%
CPU max MHz:                          4457.0000
CPU min MHz:                          400.0000
BogoMIPS:                             7186.75
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local user_shstk clzero irperf xsaveerptr rdpru wbnoinvd cppc arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm debug_swap
L1d cache:                            192 KiB (6 instances)
L1i cache:                            192 KiB (6 instances)
L2 cache:                             3 MiB (6 instances)
L3 cache:                             16 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-11
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Vulnerable: Safe RET, no microcode
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Retpolines; IBPB conditional; IBRS_FW; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected
Versions of relevant libraries:
[pip3] No relevant packages
[conda] Could not collect ROCM Version: Could not collect
Aphrodite Version: N/A
Aphrodite Build Flags:
CUDA Archs: Not Set; ROCm: Disabled

```

### How did you install Aphrodite?

```sh
pip install -U aphrodite-engine --extra-index-url https://downloads.pygmalion.chat/whl
```
```sh
Looking in indexes: https://pypi.org/simple, https://downloads.pygmalion.chat/whl
Collecting aphrodite-engine
  Using cached https://downloads.pygmalion.chat/whl/aphrodite-engine/aphrodite-engine-0.5.3.zip (1.6 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Discarding https://downloads.pygmalion.chat/whl/aphrodite-engine/aphrodite-engine-0.5.3.zip (from https://downloads.pygmalion.chat/whl/aphrodite-engine/): Requested aphrodite-engine from https://downloads.pygmalion.chat/whl/aphrodite-engine/aphrodite-engine-0.5.3.zip has inconsistent version: expected '0.5.3', but metadata has '0.5.3+cu120'
  Using cached https://downloads.pygmalion.chat/whl/aphrodite-engine/aphrodite-engine-0.5.3.tar.gz (1.3 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Discarding https://downloads.pygmalion.chat/whl/aphrodite-engine/aphrodite-engine-0.5.3.tar.gz (from https://downloads.pygmalion.chat/whl/aphrodite-engine/): Requested aphrodite-engine from https://downloads.pygmalion.chat/whl/aphrodite-engine/aphrodite-engine-0.5.3.tar.gz has inconsistent version: expected '0.5.3', but metadata has '0.5.3+cu120'
ERROR: Could not find a version that satisfies the requirement aphrodite-engine (from versions: 0.5.3)
ERROR: No matching distribution found for aphrodite-engine

**Comment by AlpinDale** [comment]
Looks like you're on python 3.12. At the moment, we only have wheels for python versions 3.8-3.11. Please downgrade. You can either use conda to do this, or a python version manager like pyenv or asdf-vm.

**Comment by calz1** [comment]
Using pyenv fixed it for 3.11.9 fixed it, thanks!
